crates/matrix-sdk-sqlite/src/utils.rs

```diff
@@ -145,6 +147,10 @@ impl SqliteObjectExt for deadpool_sqlite::Object {
         .await
         .unwrap()
     }
+
+    async fn limit(&self, limit: rusqlite::limits::Limit) -> i32 {
+        self.interact(move |conn| conn.limit(limit)).await.expect("Failed to fetch limit")
+    }
```

Comment:
hmm, the code that uses `Limit` in `chunk_large_query_over` is wrong! It casts the constant to a `i32` directly instead of calling `conn.limit` as you do here (https://github.com/matrix-org/matrix-rust-sdk/blob/18b6387a7ad8d24ca258a4ec92d6d4f7f20fa4e5/crates/matrix-sdk-sqlite/src/state_store.rs#L1624).

Maybe you can refactor your code with mine (I was the author of `chunk_large_query_over` and `repeat_vars`), fix the bug, and mutualise the tests. Thouhts?

Reply:
Great, thank you for spotting this - done in https://github.com/matrix-org/matrix-rust-sdk/pull/2945 , https://github.com/matrix-org/matrix-rust-sdk/pull/2946 and https://github.com/matrix-org/matrix-rust-sdk/pull/2947 .